src/test/java/streams/part1/exercise/Exercise1.java

```diff
@@ -0,0 +1,128 @@
+package streams.part1.exercise;
+
+import lambda.data.Employee;
+import lambda.data.JobHistoryEntry;
+import lambda.data.Person;
+import org.junit.Test;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertEquals;
+
+@SuppressWarnings({"ConstantConditions", "unused"})
+public class
+Exercise1 {
+
+    @Test
+    public void findPersonsEverWorkedInEpam() {
+        List<Employee> employees = getEmployees();
+
+        // TODO реализация, использовать Collectors.toList()
+        List<Person> personsEverWorkedInEpam = employees.stream().
+                filter(employee -> employee.getJobHistory().
+                        stream().
+                        anyMatch(job -> job.getEmployer().equals("EPAM"))).
+                map(Employee::getPerson).
+                collect(Collectors.toList());
+
+        List<Person> expected = Arrays.asList(
+                employees.get(0).getPerson(),
+                employees.get(1).getPerson(),
+                employees.get(4).getPerson(),
+                employees.get(5).getPerson());
+        assertEquals(expected, personsEverWorkedInEpam);
+    }
+
+    @Test
+    public void findPersonsBeganCareerInEpam() {
+        List<Employee> employees = getEmployees();
+
+        // TODO реализация, использовать Collectors.toList()
+        List<Person> startedFromEpam = employees.stream().
+                filter(employee -> employee.getJobHistory().get(0).getEmployer().equals("EPAM")).
+                map(Employee::getPerson).
+                collect(Collectors.toList());
+
+        List<Person> expected = Arrays.asList(
+                employees.get(0).getPerson(),
+                employees.get(1).getPerson(),
+                employees.get(4).getPerson());
+        assertEquals(expected, startedFromEpam);
+    }
+
+    @Test
+    public void findAllCompanies() {
+        List<Employee> employees = getEmployees();
+
+        // TODO реализация, использовать Collectors.toSet()
+        Set<String> companies = employees.stream().
+                map(employee -> employee.getJobHistory().stream()).
+                flatMap(jobHistoryEntryStream -> jobHistoryEntryStream.map(JobHistoryEntry::getEmployer)).
+                map(String::toString).
```

Comment:
Maybe I am wrong, but are you sure, that you need this? At `line 61` you receive `employer` as `String`.

Reply:
fixed 94ab8bda1d601b868ea1b089dca2b9c156671659